lib/testing.dart

```diff
@@ -0,0 +1,133 @@
+import 'dart:async';
+
+import 'package:web_http/web_http.dart';
+
+/// A signature for callback which is invoked whenever the [TestHttpClient]
+/// recieves a request.
+typedef FutureOr<TestHttpResponse> TestClientCallback(String url, String method,
+    {String responseType,
+    String body,
+    Map<String, String> headers,
+    int timeout,
+    bool withCredentials});
+
+/// A testing implementation of [HttpClient].
+/// 
+/// The exact responses returned can be controlled with the provided 
+/// [callback].
+class TestHttpClient implements HttpClient {
```

Comment:
Would it be a good idea to extend rather than implement `HttpClient` so you don't need to re-implement `get` and `post`?

Reply:
Not that they're much work to re-implement, but is there an advantaged to implementing over extending?

Reply:
Yeah, this seems like a brainfart on my part - fixed!